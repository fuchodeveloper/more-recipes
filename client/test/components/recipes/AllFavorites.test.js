/* eslint-disable max-len */
import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { AllFavoriteRecipes }
  from '../../../components/favorites/AllFavoriteRecipes';

let props;

const setup = () => {
  props = {
    details: {
      Recipe: {
        image: 'https://res.cloudinary.com/fuchodeveloper/image/upload/v1516760699/noodles_c6ltkq.jpg',
        name: 'jollof',
        direction: 'rice, beans',
        views: 0,
        id: 1
      }
    }
  };
  return shallow(<AllFavoriteRecipes {...props} />);
};

describe('AllFavoriteRecipes component', () => {
  it('should render with right amount of elements', () => {
    expect(setup()).toMatchSnapshot();
  });
});

