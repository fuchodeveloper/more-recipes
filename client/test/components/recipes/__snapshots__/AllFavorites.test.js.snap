// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllFavoriteRecipes component should render with right amount of elements 1`] = `
<div
  className="col-md-4 mb-5"
>
  <div
    className="card mt-1"
  >
    <img
      alt="jollof"
      className="card-img-top"
      height="231px"
      max-width="348px"
      src="https://res.cloudinary.com/fuchodeveloper/image/upload/v1516760699/noodles_c6ltkq.jpg"
    />
    <div
      className="card-body"
    >
      <h4
        className="card-title favorite-recipe-title"
      >
        <strong>
          jollof
        </strong>
      </h4>
      <p
        className="card-text recipe-font-family"
      >
        rice, beans...
      </p>
      <div
        className="card-footer custom-card-footer-bg"
      >
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
            0
            <i
              aria-hidden="true"
              className="fa fa-eye"
            />
             .
            <i
              aria-hidden="true"
              className="fa fa-thumbs-up"
            />
             .
            <i
              aria-hidden="true"
              className="fa fa-star"
            />
          </small>
        </p>
        <Link
          className="btn btn-primary btn-primary-color"
          replace={false}
          to="/recipes/1"
        >
          View Recipe
        </Link>
      </div>
    </div>
  </div>
</div>
`;
